src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -93,6 +97,7 @@ public string Path
                 throw new ArgumentNullException(nameof(context));
             }
 
+            _bufferSource = context.HttpContext.RequestServices.GetRequiredService<IRazorBufferSource>();
```

Comment:
I'm still not following this - why is the same view instance being executed from multiple threads?